search/search-client-elasticsearch/src/main/java/io/camunda/search/es/transformers/aggregator/SearchFiltersAggregatorTransformer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.es.transformers.aggregator;
+
+import co.elastic.clients.elasticsearch._types.aggregations.Aggregation;
+import co.elastic.clients.elasticsearch._types.aggregations.Buckets;
+import co.elastic.clients.elasticsearch._types.aggregations.FiltersAggregation;
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import io.camunda.search.clients.aggregator.SearchFiltersAggregator;
+import io.camunda.search.es.transformers.ElasticsearchTransformers;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+public class SearchFiltersAggregatorTransformer
+    extends AggregatorTransformer<SearchFiltersAggregator, Aggregation> {
+
+  public SearchFiltersAggregatorTransformer(final ElasticsearchTransformers transformers) {
+    super(transformers);
+  }
+
+  @Override
+  public Aggregation apply(final SearchFiltersAggregator value) {
+    final var queries =
+        value.queries().entrySet().stream()
+            .collect(
+                Collectors.toMap(Entry::getKey, e -> getQueryTransformer().apply(e.getValue())));
+    final var filterBuckets =
+        new FiltersAggregation.Builder()
+            .filters(new Buckets.Builder<Query>().keyed(queries).build())
+            .build();
+
+    final var builder = new Aggregation.Builder().filters(filterBuckets);
+    applySubAggregations(builder, value);
```

Comment:
💭  applySubAggregations(builder, value); looks fine, just ensure it’s not mutating anything unexpectedly (based on how Aggregation.Builder behaves – I assume it’s safe here).  💯

Reply:
I was also thinking about having a method that doesn't even need the builder and just returns the map of the sub-aggregations.
I decided for this approach because this way we don't need to check if the returned aggregations are empty or not and then assign them. A bit less code in the actual aggregators.
But yes, the `applySubAggregations` only calls the `builder.aggregations()` method and it shouldn't do more.

Reply:
Thanks for clarifying :D